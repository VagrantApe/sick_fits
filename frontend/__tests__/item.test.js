import ItemComponent from "../components/Item";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

const fakeItem = {
  id: "ABC124",
  title: "Sweet Item",
  price: 4000,
  desription: "Test Item",
  image: "item.jpg",
  largeImage: "largeItem.jpg"
};

describe("<Item />", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<ItemComponent item={fakeItem} />);
  });
  it("renders and matches the snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  // it("renders the image properly", () => {
  //   const img = wrapper.find("img");
  //   console.log(img.props());
  //   expect(img.props().alt).toBe(fakeItem.title);
  //   expect(img.props().src).toBe(fakeItem.image);
  // });
  // it("renders and displays properly", () => {
  //   const PriceTag = wrapper.find("PriceTag");
  //   console.log(PriceTag.debug());
  //   console.log(PriceTag.children().text());
  //   expect(PriceTag.children().text()).toBe("$50");
  //   expect(wrapper.find("Title a").text()).toBe(fakeItem.title);
  // });
  // it("renders out buttons properly", () => {
  //   const buttonList = wrapper.find(".buttonList");
  //   expect(buttonList.children()).toHaveLength(3);
  //   expect(buttonList.find("Link").exists()).toBe(true);
  //   expect(buttonList.find("AddToCart").exists()).toBe(true);
  //   expect(buttonList.find("DeleteItem").exists()).toBe(true);
  // });
});
