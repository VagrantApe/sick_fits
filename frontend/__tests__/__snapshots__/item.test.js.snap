// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="Sweet Item"
    src="item.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "ABC124",
          },
        }
      }
    >
      <a>
        Sweet Item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $40
  </PriceTag>
  <p />
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "/update",
          "query": Object {
            "id": "ABC124",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="ABC124"
    />
    <DeleteItem
      id="ABC124"
    >
      Delete Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
